packages/go_router/lib/src/builder.dart

```diff
@@ -594,7 +594,14 @@ class _PagePopContext {
     final Page<Object?> page = route.settings as Page<Object?>;
 
     final RouteMatch match = _routeMatchesLookUp[page]!.last;
-    _routeMatchesLookUp[page]!.removeLast();
+
+    // If a popup is opened using something like Scaffold.showBottomSheet,
+    // it will be registered in LocalHistoryRoute._localHistory,
+    // so it should be deleted from _routeMatchesLookUp only if
+    // LocalHistoryRoute.willHandlePopInternally is false.
+    if (route is LocalHistoryRoute && route.willHandlePopInternally == false) {
```

Comment:
I am a bit confused, if willHandlePopInternally is true, it should not call this function at the first place

Reply:
It says "it should not call this function at the first place", but if you close the Bottom Sheet displayed with showBottomSheet by calling "Navigator.pop(context);", go_router will end up here.